docs/how-to-integrate-a-frontend.md

```diff
@@ -0,0 +1,247 @@
+
+Calling the contract requires the following steps:
+
+- Create a new contract instance via the `.next()` method of the current instance. Update the state of the new instance to the latest.
+- Add a transaction builder to the `move()` method of the `TicTacToe` contract through the `bindTxBuilder` method, which builds the calling transaction.
+- Finally, call the `methods` public method on the contract instance to send the transaction to call the contract. Every public method `xxx` has a function of the same name under `contractInstance.methods.xxx`. It takes the same parameters, plus a `MethodCallOptions`.
+
+If the public method has a signature parameter, a callback function is required to return the signature. The `Signer` connected to the contract will use the default private key to sign, and the signature will be returned through the parameter `sigResps` of the callback function. Use `findSig()` to find the signature associated with a public key.
+
+Through `pubKeyOrAddrToSign` in `MethodCallOptions`, you can specify which address/public key to sign against.
+
+```ts
+const { tx: callTx } = await p2pkh.methods.unlock(
+    (sigResponses: SignatureResponse[]) => findSig(sigResponses, $publickey),
+    $publickey,
+    {
+        pubKeyOrAddrToSign: $publickey.toAddress()
+    } as MethodCallOptions<P2PKH>
+);
+```
+
+- After the call is completed, the new contract instance needs to be saved, to continue calling the contract later.
+
+The code implementation of the above steps:
+
+```ts
+// 1. create nextInstance
+const current = props.contract as TicTacToe;
+const nextInstance = current.next();
+// convert latest game data to contract state with Utils.toContractState and update nextInstance state
+Object.assign(nextInstance, Utils.toContractState(latestGameData));
+
+// 2. bind a tx builder for move
+TicTacToe.bindTxBuilder('move', async (options: BuildMethodCallTxOptions<SmartContract>, n: bigint, sig: Sig) => {
+    ...
+}
+
+// 3. call contract.methods.move(...) to broadcast transaction
+const {tx, next} = await current.methods.move(
+    BigInt(i),
+    (sigResponses: SignatureResponse[]) => findSig(sigResponses, $publickey)
+);
+
+// 4. save latest contract instance
+props.setContract(next?.instance)
+```
+
+So far, we have completed the interaction between the `TicTacToe` contract and the web frondend. Every move will prompt the player to sign and submit a transaction on the blockchain.
+
+![](../static/img/play-tic-tac-toe.gif)
+
+You can visit the full complete code of [`Game.tsx`](https://github.com/sCrypt-Inc/tic-tac-toe/blob/main/src/Game.tsx) for more details.
+
+## Conclusion
+
+Great, you made it to the end of this last part! Now you're fully equipped to apply the skills from this tutorial to build out your own custom dApp project!
+
+The full complete code of this demo is at [Bitcoin on-chain App Tutorial](https://github.com/sCrypt-Inc/tic-tac-toe).
```

Comment:
Link to learn, not github https://learn.scrypt.io/en/courses/Build-a-Tic-tac-toe-Game-with-sCrypt-614c387bc0974f55df5af1e5